### PR TITLE
Change the Concrete KL to logit space to avoid underflow (fixed)

### PR DESCRIPTION
**Pull request by eelcovdw**
Madison et al. recommend using the logits of the concrete distribution as stochastic node, to avoid numerical underflow in the KL. Pytorch now supports this, so we only need to change the KL estimation to use the base LogitRelaxedBernoulli distribution instead of the RelaxedBernoulli.

Small fix from PR #3: use sample instead of rsample.

### CODE REVIEW

**Comment by wilkeraziz** [comment]
Why should we use sample rather than rsample from base_dist?

**Comment by eelcovdw** [comment]
Good point, rsample is correct here after all. In that case, #3 is correct, and this one should be closed.
